packages/blade/src/hooks/useExposeInputRef.ts

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import type { TextInput as TextInputReactNative } from 'react-native';
+
+type NativeHTMLInputRefProps = Pick<HTMLElement, 'focus' | 'scrollIntoView'>;
+
+/**
+ * A hook which only exposes the properties of html input element via imparative hook
+ *
+ * It avoids exposing other native properties of HTMLElement
+ * like `style` `classList` to avoid unintended usage of refs.
+ */
+const useExposeInputRef = (
```

Comment:
its not limited to input ref. it might be used on button, checkbox, radio, etc. so maybe we can rename it to `useMaskedRef` or `useSafeRef` or if you can think about better name

Reply:
🤔 but it's only for inputs tho, checkbox/radios are just basically <inputs \/> and button will not be exposed like that with the ref right?

Reply:
why not?

Reply:
wait are we exposing refs for Button component too?

Reply:
yes we have to.